src/core/thread/mle.cpp

```diff
@@ -4229,6 +4234,52 @@ uint64_t Mle::CalcParentCslMetric(const Mac::CslAccuracy &aCslAccuracy) const
 }
 #endif
 
+#if OPENTHREAD_CONFIG_WAKEUP_COORDINATOR_ENABLE
+void Mle::HandleWedAttachTimer(void)
+{
+    switch (mWedAttachState)
+    {
+    case kWedAttaching:
+        // Connection timeout
+        mWedAttachState = kWedDetached;
+
+        if (!IsRxOnWhenIdle())
+        {
+            Get<MeshForwarder>().SetRxOnWhenIdle(false);
+        }
+
+        LogInfo("Connection window closed");
+        break;
+    default:
+        break;
+    }
+}
+
+Error Mle::AttachWed(const Mac::ExtAddress &aWedAddress, uint16_t aIntervalUs, uint16_t aDurationMs)
+{
+    Error error;
+
+    VerifyOrExit(mWedAttachState == kWedDetached, error = kErrorInvalidState);
```

Comment:
Should we check `(aIntervalUs  < aDurationMs)  && (aIntervalUs != 0) && (aDurationMs != 0)` here?